integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkApplicationExecutionContext.java

```diff
@@ -137,7 +137,17 @@ private OpenLineage.ParentRunFacet buildApplicationParentFacet() {
       return ol.newParentRunFacet(
           ol.newParentRunFacetRun(eventEmitter.getParentRunId().get()),
           ol.newParentRunFacetJob(
-              eventEmitter.getParentJobNamespace().get(), eventEmitter.getParentJobName().get()));
+              eventEmitter.getParentJobNamespace().get(), eventEmitter.getParentJobName().get()),
+          ol.newParentRunFacetRoot(
+              ol.newRootRun(
+                  eventEmitter.getRootParentRunId().orElse(eventEmitter.getParentRunId().get())),
```

Comment:
The code assumes `eventEmitter.getParentRunId().get() will always be valid, but if `getRootParentRunId()` is empty and getParentRunId() is also empty, wouldn't this throw an exception?

Reply:
Seems inconsistent too:

In both `RddExecutionContext.buildApplicationParentFacet()` and `SparkSQLExecutionContext.buildApplicationParentFacet()`, the fallback chain is:

```
eventEmitter
    .getRootParentRunId()
    .orElse(eventEmitter.getParentRunId().orElse(eventEmitter.getApplicationRunId()))
```
But in `SparkApplicationExecutionContext.buildApplicationParentFacet()`, there's no fallback to `applicationRunId`.

Reply:
Because SparkApplication is parent for RDDContext and SparkSQL, but not it's own parent